src/arch/x86_64.rs

```diff
@@ -131,15 +139,27 @@ impl hal_core::interrupt::Handlers<X64Registers> for InterruptHandlers {
     where
         C: hal_core::interrupt::ctx::Context<Registers = X64Registers>,
     {
-        tracing::info!(registers=?cx.registers(), "lol im in ur test interrupt");
+        let fired = TEST_INTERRUPT_WAS_FIRED.fetch_add(1, Ordering::Release) + 1;
+        tracing::info!(registers = ?cx.registers(), fired, "lol im in ur test interrupt");
     }
 }
 
-#[no_mangle]
 #[cfg(target_os = "none")]
-pub extern "C" fn _start(info: &'static bootinfo::BootInfo) -> ! {
-    let bootinfo = RustbootBootInfo { inner: info };
-    crate::kernel_main(&bootinfo);
+bootloader::entry_point!(arch_entry);
+
+pub fn arch_entry(info: &'static mut boot_info::BootInfo) -> ! {
+    if let Some(offset) = info.physical_memory_offset.into_option() {
+        // Safety: i hate everything
+        unsafe {
+            vga::init_with_offset(offset);
+        }
+    }
```

Comment:
I'm not sure I understand why we need to force-unlock the VGA buffer here. Why would it be locked?

Reply:
i don't think it's necessary, this was just scattershot debugging